src/ripple/server/impl/BaseWSPeer.h

```diff
@@ -394,10 +462,18 @@ on_timer(error_code ec)
 {
     if(ec == boost::asio::error::operation_aborted)
         return;
-    if(! ec)
+
+    if (! ec && timer_action_ != 0)
```

Comment:
We want to make sure that we are able to disconnect the connection if two timers trigger without us receiving a pong. It is at least theoretically possible that we could get stuck trying to send the ping and we don't want that to keep the connection around forever.